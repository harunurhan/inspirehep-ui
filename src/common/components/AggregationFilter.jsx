import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Immutable from 'immutable';

import CheckboxAggregation from './CheckboxAggregation';
import RangeAggregation from './RangeAggregation';

class AggregationFilter extends Component {
  render() {
    const {
      buckets,
      onChange,
      name,
      range,
      selections,
    } = this.props;

    if (range) {
      return (
        <RangeAggregation
          name={name}
          buckets={buckets}
          onChange={onChange}
          selections={selections}
        />
      );
    } else {
      return (
        <CheckboxAggregation
          name={name}
          buckets={buckets}
          onChange={onChange}
          selections={selections}
        />
      );
    }
  }
}

AggregationFilter.propTypes = {
  onChange: PropTypes.func.isRequired,
  buckets: PropTypes.instanceOf(Immutable.List).isRequired,
  name: PropTypes.string.isRequired,
  selections: PropTypes.arrayOf(PropTypes.any),
  range: PropTypes.bool,
};

AggregationFilter.defaultProps = {
  range: false,
  selections: [],
};

export default AggregationFilter;
