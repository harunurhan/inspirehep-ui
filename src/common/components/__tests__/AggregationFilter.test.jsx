import React from 'react';
import { List, fromJS } from 'immutable';
import { shallow } from 'enzyme';

import AggregationFilter from '../AggregationFilter';

describe('AggregationFilter', () => {
  it('render initial state with all props set', () => {
    const buckets = fromJS([
      {
        key: 'bucket1',
        doc_count: 1,
      },
      {
        key: 'bucket2',
        doc_count: 2,
      },
    ]);
    const wrapper = shallow(<AggregationFilter
      onChange={jest.fn()}
      buckets={buckets}
      name="Test"
      selectedKeys={['bucket1']}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('derives selectionMap state from prop selectedKeys', () => {
    const selectedKeys = ['selected1', 'selected2'];
    const wrapper = shallow(<AggregationFilter
      onChange={jest.fn()}
      buckets={List()}
      name="Test"
      selectedKeys={selectedKeys}
    />);
    const { selectionMap } = wrapper.instance().state;
    expect(selectionMap.get('selected1')).toBe(true);
    expect(selectionMap.get('selected2')).toBe(true);
  });

  describe('onBucketChange', () => {
    it('calls onChange with all selections', () => {
      const onChange = jest.fn();
      const wrapper = shallow(<AggregationFilter
        onChange={onChange}
        buckets={List()}
        name="Test"
      />);
      wrapper.instance().onBucketChange('selected1', true);
      expect(onChange).toHaveBeenCalledWith(['selected1']);
      wrapper.instance().onBucketChange('selected2', true);
      expect(onChange).toHaveBeenCalledWith(['selected1', 'selected2']);
    });
  });
});
