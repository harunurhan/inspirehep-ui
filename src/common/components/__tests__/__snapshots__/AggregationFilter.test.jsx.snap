// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregationFilter renders CheckboxAggregation if range prop is not set 1`] = `
<CheckboxAggregation
  buckets={
    Immutable.List [
      Immutable.Map {
        key: "bucket1",
        doc_count: 1,
      },
      Immutable.Map {
        key: "bucket2",
        doc_count: 2,
      },
    ]
  }
  name="Test"
  onChange={[Function]}
  selections={
    Array [
      "bucket1",
    ]
  }
/>
`;

exports[`AggregationFilter renders RangeAggregation if range prop is true 1`] = `
<RangeAggregation
  buckets={
    Immutable.List [
      Immutable.Map {
        key: "bucket1",
        doc_count: 1,
      },
      Immutable.Map {
        key: "bucket2",
        doc_count: 2,
      },
    ]
  }
  countPropName="doc_count"
  deselectedColor="#fff"
  height={100}
  hoverColor="#69c0ff"
  keyPropName="key_as_string"
  name="Test"
  onChange={[Function]}
  selectedColor="#91d5ff"
  selections={
    Array [
      "bucket1",
    ]
  }
  width={200}
/>
`;
